starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -51,8 +51,10 @@ public List<Inventory> findAll() {
    * @return Created/Updated Inventory.
    */
   public Inventory create(Inventory inventory) {
-    // TODO
-    return null;
+    //Inserts inventory and changes Id
+    this.mongoTemplate.insert(inventory);
+    inventory.setId(null);
```

Comment:
So, the goal here I believe is to create a new inventory with a new ID. The reason that the task asks you to make the ID null is to ensure that this is a new inventory we are creating and that we are not updating an already existing inventory.  (Because mongo will check if there is an inventory with that ID already existing in the database before inserting something identical. You want to ensure that mongo will create a new inventory in the database no matter what.) 
Making the ID null _after_ inserting into the database basically just results in an inventory item returned with a null ID. What you actually want to do is create a new inventory and let mongo give it a unique ID. 

I remember this part of the task being worded weirdly, so feel free to ask if what I said is confusing.